auth/forward.go

```diff
@@ -58,3 +62,32 @@ func Forward(forward *types.Forward, w http.ResponseWriter, r *http.Request, nex
 	r.RequestURI = r.URL.RequestURI()
 	next(w, r)
 }
+
+func writeHeader(req *http.Request, forwardReq *http.Request) {
+	utils.CopyHeaders(forwardReq.Header, req.Header)
+
+	if clientIP, _, err := net.SplitHostPort(req.RemoteAddr); err == nil {
+		if prior, ok := req.Header[forward.XForwardedFor]; ok {
+			clientIP = strings.Join(prior, ", ") + ", " + clientIP
+		}
+		forwardReq.Header.Set(forward.XForwardedFor, clientIP)
+	}
+
+	if xfp := req.Header.Get(forward.XForwardedProto); xfp != "" {
+		forwardReq.Header.Set(forward.XForwardedProto, xfp)
+	} else if req.TLS != nil {
+		forwardReq.Header.Set(forward.XForwardedProto, "https")
+	} else {
+		forwardReq.Header.Set(forward.XForwardedProto, "http")
+	}
+
+	if xfp := req.Header.Get(forward.XForwardedPort); xfp != "" {
+		forwardReq.Header.Set(forward.XForwardedPort, xfp)
+	}
+
+	if xfh := req.Header.Get(forward.XForwardedHost); xfh != "" {
```

Comment:
If a client access to `admin.domain.local` and send the header X-FORWARDED-HOST ` user.domain.local` he can by-pass the auth, no ?

Reply:
yes, I added a new option "trustForwardHeader"